### PR TITLE
Updated categorical_hinge loss docs in losses.py

### LINKED ISSUE #38199: tf.keras.losses.categorical_hinge mentions [-1, 1] values while it works with one-hot-encoded tensor
Hi!

Please see:
https://www.tensorflow.org/api_docs/python/tf/keras/losses/categorical_hinge  
https://github.com/tensorflow/tensorflow/blob/v2.1.0/tensorflow/python/keras/losses.py#L866-L882

Both mention:
> y_true: The ground truth values. y_true values are expected to be -1 or 1. If binary (0 or 1) labels are provided they will be converted to -1 or 1.

While the code is --as expected-- a transcription of keras' one:
```
# ...
y_pred = ops.convert_to_tensor(y_pred)
y_true = math_ops.cast(y_true, y_pred.dtype)
pos = math_ops.reduce_sum(y_true * y_pred, axis=-1)
neg = math_ops.reduce_max((1. - y_true) * y_pred, axis=-1)
return math_ops.maximum(0., neg - pos + 1.)
```

And this code is meant to work with one-hot-encoded tensors. See the original discussion here: https://github.com/keras-team/keras/issues/2830

### PR DESCRIPTION
**Pull request by ManishAradwad**
Fixes : #38199

### CODE REVIEW

File: tensorflow/python/keras/losses.py [hunk 1]
```diff
@@ -1401,7 +1401,8 @@ def categorical_hinge(y_true, y_pred):
   >>> assert np.array_equal(loss.numpy(), np.maximum(0., neg - pos + 1.))
 
   Args:
-    y_true: The ground truth values. `y_true` values are expected to be 0 or 1.
+    y_true: The ground truth values. `y_true` values are expected to be
+    either {-1, +1} or {0, 1}(i.e. a one-hot-encoded tensor).
```
Comment (fchollet) [inline tensorflow/python/keras/losses.py:1405]: Please use backticks for code formatting.

**Comment by gbaned** [comment]
@ManishAradwad  Can you please check @fchollet's comments and keep us posted ? Thanks!

**Comment by gbaned** [comment]
@ManishAradwad  Any update on this PR? Please. Thanks!

**Comment by ManishAradwad** [comment]
Hello @gbaned,
Sorry for not being in touch. At this moment, I won't be able to work on this one for a while. Please, feel free to assign it to someone else.
Thanks!

File: tensorflow/python/keras/losses.py (hunk 1 above)
Comment (rmothukuru) [reply tensorflow/python/keras/losses.py:1405]: @fchollet,
Done

**Review by mihaimaruseac — APPROVED** [review]

File: tensorflow/python/keras/losses.py [hunk 2]
```diff
@@ -1403,19 +1403,22 @@ def categorical_hinge(y_true, y_pred):
...
+  ```python
+  y_true = np.random.randint(0, 3, size=(2,))
+  y_true = tf.keras.utils.to_categorical(y_true, num_classes=3)
+  y_pred = np.random.random(size=(2, 3))
+  loss = tf.keras.losses.categorical_hinge(y_true, y_pred)
+  assert loss.shape == (2,)
+  pos = np.sum(y_true * y_pred, axis=-1)
+  neg = np.amax((1. - y_true) * y_pred, axis=-1)
+  assert np.array_equal(loss.numpy(), np.maximum(0., neg - pos + 1.))
+  ```
```
Comment (mihaimaruseac) [inline tensorflow/python/keras/losses.py:1417]: This should stay as a doctest format (with `>>> `)

File: tensorflow/python/keras/losses.py [hunk 3]
```diff
@@ -1403,19 +1403,22 @@ def categorical_hinge(y_true, y_pred):
...
-  >>> loss = tf.keras.losses.categorical_hinge(y_true, y_pred)
-  >>> assert loss.shape == (2,)
-  >>> pos = np.sum(y_true * y_pred, axis=-1)
-  >>> neg = np.amax((1. - y_true) * y_pred, axis=-1)
-  >>> assert np.array_equal(loss.numpy(), np.maximum(0., neg - pos + 1.))
 
   Args:
-    y_true: The ground truth values. `y_true` values are expected to be 0 or 1.
+    y_true: The ground truth values. `y_true` values are expected to be
+    either {-1, +1} or {0, 1}(i.e. a one-hot-encoded tensor).
```
Comment (mihaimaruseac) [inline tensorflow/python/keras/losses.py:1421]:
```suggestion
    either `{-1, +1}` or `{0, 1}` (i.e. a one-hot-encoded tensor).
```

File: tensorflow/python/keras/losses.py (hunk 2 above)
Comment (rmothukuru) [reply tensorflow/python/keras/losses.py]: Replaced Backticks (```) with >>>

File: tensorflow/python/keras/losses.py (hunk 3 above)
Comment (rmothukuru) [reply tensorflow/python/keras/losses.py]: Done Mihai

**Review by gbaned — APPROVED** [review]

**Review by mihaimaruseac — APPROVED** [review]
